tempesta_fw/ss_skb.c

```diff
@@ -313,8 +368,17 @@ __split_linear_data(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 	 */
 	if (unlikely((len < 0) && (tail_len == -len))) {
 		ss_skb_put(skb, len);
-		if (skb_is_nonlinear(skb))
+		if (skb_is_nonlinear(skb)) {
 			it->ptr = skb_frag_address(&skb_shinfo(skb)->frags[0]);
+			it->skb = pskb ? pskb : skb;
+		} else {
+			struct sk_buff *dskb = __next_skb_frag(skb, pskb);
```

Comment:
You've just checked ```pskb``` in ```__next_skb_frag()``` and checks it again here. That's clumsy. Moreover exactly the same piece of code is copy-pasted in ```__split_pgfrag_del()```. Thus the piece of code must be inline function not using ```__next_skb_frag()```, but rather accurately implementing the logic with minimum branches. Also, ```a ? a : b``` usually is expressed as ```a ? : b```.